File: src/domain/Comment.java
package domain;

public class Comment {
    private Integer blogId = 0;
    private String name;
    private String content;
    private Integer rating;

    public Comment() {}

    public Comment(String name, String content, Integer rating) {
        setName(name);
        setContent(content);
        setRating(rating);
    }

    public Comment(Integer blogId, String name, String content, Integer rating) {
        setBlogId(blogId);
        setName(name);
        setContent(content);
        setRating(rating);
    }

    public Integer getBlogId() {
        return this.blogId;
    }

    public void setBlogId(Integer blogId) {
        this.blogId = blogId;
    }

    public String getName() {
        return this.name;
    }

    public void setName(String name) {
        if (name.isEmpty())
            throw new IllegalArgumentException("Name should not be empty");

        this.name = name;
    }

    public String getContent() {
        return this.content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Integer getRating() {
        return this.rating;
    }

    public void setRating(Integer rating) {
        if (rating < 1 || rating > 10)
            throw new IllegalArgumentException("Rating should be between 1 and 10");

        this.rating = rating;
    }
}

File: web/js/comment.js
var webSocket = new WebSocket("ws://localhost:8080/comment");

webSocket.onmessage = function (event) {
    var data = JSON.parse(event.data);
    data["name"] = escapeHtml(data["name"]);
    data["content"] = escapeHtml(data["content"]);

    var blogPostMessage = document.getElementById("blogpost-" + data["blogId"]).querySelectorAll('[class=message-body]')[0];
    blogPostMessage.innerHTML += "<strong>" +
        data["name"] +
        ": </strong>" +
        data["content"] +
        " <span class=\"icon is-small\"><i class=\"fas fa-star\"></i></span> " +
        data["rating"] +
        "<br />"
};

function postComment(id) {
    var blogPost = document.getElementById("blogpost-" + id);
    var data = {};
    data["blogId"] = id;
    data["name"] = blogPost.querySelectorAll('[name=name]')[0].value;
    data["content"] = blogPost.querySelectorAll('[name=comment]')[0].value;
    data["rating"] = parseInt(blogPost.querySelectorAll('[name=rating]')[0].value);

    validComment(data);

    webSocket.send(JSON.stringify(data));
}

function validComment(data) {
    if (data["name"].length < 1) {
        alert("Name can not be empty");
    }

    if (data["rating"] < 1 || data["rating"] > 10) {
        alert("Rating should be between 1 and 10");
    }
}

function escapeHtml(unsafe) {
    return unsafe
        .replace(/&/g, "&amp;")
        .replace(/</g, "&lt;")
        .replace(/>/g, "&gt;")
        .replace(/"/g, "&quot;")
        .replace(/'/g, "&#039;");
}